log4j-api/src/main/java/org/apache/logging/log4j/util/PropertiesPropertySource.java

```diff
@@ -29,15 +29,21 @@
 
     private static final String PREFIX = "log4j2.";
 
+    private static final String KEY_PRIORITY = "log4j2.properties_load_priority";
+
+    private final int priority;
+
     private final Properties properties;
 
     public PropertiesPropertySource(final Properties properties) {
         this.properties = properties;
+        final String order = (String)properties.get(KEY_PRIORITY);
```

Comment:
Log4j has a fuzzy matching strategy, I am not sure if we can simply look for `KEY_PRIORITY` at this stage – the fuzzy key alternatives might not have been populated.